### PR TITLE
ACM-13470: upgrade hive in clc repos

### PR DESCRIPTION
**Pull request by dtthuynh**
https://issues.redhat.com/browse/ACM-13470

### CODE REVIEW

**Comment by dtthuynh** [comment]
@fxiang1 thanks :) the unit-tests log does indicate its using 1.21 image. I opened a PR to update the 2.7/main prow files for this and the cluster-curator repo https://github.com/openshift/release/pull/56245

**Comment by dtthuynh** [comment]
We need to update the controller-runtime lib in order to work after upgrading the go version.
refer https://github.com/kubernetes-sigs/controller-runtime/issues/2788

**Comment by dtthuynh** [comment]
Noting here, the unit test failing is due to a change in newer controller-runtime which leaves TypeMeta empty, causing the unit test on the mc annotation to fail (found by @fxiang1 ): https://github.com/kubernetes-sigs/controller-runtime/issues/1517

cc @xuezhaojun

File: controllers/managedcluster/managedcluster-controller_test.go [hunk 1]
```diff
@@ -28,12 +28,16 @@ func TestReconcile(t *testing.T) {
...
 
 	c := &ManagedClusterReconciler{
-		Client: clientfake.NewFakeClientWithScheme(testScheme),
+		Client: clientfake.NewClientBuilder().WithScheme(testScheme).Build(),
 		Log:    ctrl.Log.WithName("controllers").WithName("ManagedClusterReconciler"),
 		Scheme: testScheme,
 	}
 
 	if err := c.Client.Create(ctx, &hivev1.ClusterClaim{
+		// TypeMeta: metav1.TypeMeta{
```
Comment (dtthuynh) [inline controllers/managedcluster/managedcluster-controller_test.go:37]: @fxiang1 @xiangjingli I left this commented out so the test doesn't pass falsely, but I assume we need to define this elsewhere, maybe in the managedcluster-controller.go or clusterclaims-controller? We know the gvk via the hivev1 scheme anyway https://github.com/openshift/hive/blob/b6cdaa9cb31727f22c9e354abc321b7ee0ca4ee6/contrib/pkg/clusterpool/clusterclaim.go#L77

**Review by fxiang1 — APPROVED** [review]
Thanks David!

File: controllers/managedcluster/managedcluster-controller.go [hunk 2]
```diff
@@ -79,8 +79,8 @@ func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, request ctrl.R
 		return ctrl.Result{}, err
 	}
 
-	// annotation format is <name>.<namespace>.<kind>.<apiversion>
-	expectedProvisioner := fmt.Sprintf("%s.%s.%s.%s", claimName, claimNamespace, clusterClaim.Kind, clusterClaim.APIVersion)
+	// annotation format is <name>.<namespace>
```
Comment (zhujian7) [inline controllers/managedcluster/managedcluster-controller.go:82]: @xiangjingli the kind and apiversion should not be removed, I guess this is the root cause for https://issues.redhat.com/browse/ACM-16131

**Pull request merged**